packages/flutter/lib/src/widgets/async.dart

```diff
@@ -204,6 +204,19 @@ class AsyncSnapshot<T> {
   /// Latest data received. Is `null`, if [error] is not.
   final T data;
 
+  /// Returns latest data received, failing if there is no data.
+  ///
+  /// Throws [error], if [hasError]. Throws [StateError], if neither [hasData]
+  /// nor [hasError].
+  T get requireData {
+    if (hasData)
+      return data;
+    else if (hasError)
```

Comment:
I don't see it explicitly in the style guide, but generally we skip the `else` after return or throw statements.  (Also applies to line 216